anthos-bm-apigee/docs/quickstart.md

```diff
@@ -0,0 +1,590 @@
+#gcp_login_accounts           = ["<GCP_ACCOUNT_1>", <GCP_ACCOUNT_2>, <GCP_ACCOUNT_3>]
+username                      = tfadmin
+#gce_vm_service_account        = "<GCP VM Service Account>"
+```
+Uncomment the gce_vm_service_account if you are using custom compute engine service account
+
+An example of these configuration looks like this below:
+
+```
+project_id                     = "anthos-bm-example1"
+region                         = "us-central1"
+zone                           = "us-central1-a"
+credentials_file               = "anthos-bm-owner.json"
+username                      = tfadmin
+#gce_vm_service_account        = "apigee-admin-ws@anthos-bm-example1.iam.gserviceaccount.com"
+```
+
+4. Rename the `variables` file to default name used by Terraform for the `variables` file:
+> **Note:** You can skip this step if you run `terraform apply` with the `-var-file` flag
+```sh
+mv terraform.tfvars.sample terraform.tfvars
+```
+
+5. Navigate to the root directory of this repository initialize it as a Terraform directory
+```sh
+# this sets up the required Terraform state management configurations, similar to 'git init'
+terraform init
+```
+
+6. Create a _Terraform_ execution plan
+```sh
+# compares the state of the resources, verifies the scripts and creates an execution plan
+terraform plan
+```
+
+7. Apply the changes described in the _Terraform_ script
+```sh
+# executes the plan on the given provider (i.e: GCP) to reach the desired state of resources
+terraform apply
+```
+> **Note:** When prompted to confirm the Terraform plan, type 'Yes' and enter
+
+***The `apply` command sets up the Compute Engine VM based bare metal infrastructure. This can take a few minutes (approx. 3-5 mins) for the entire bare-metal cluster to be setup.***
+
+---
+### Deploy an Anthos cluster
+
+After the Terraform execution completes you are ready to deploy an Anthos cluster.
+
+1. SSH into the admin host
+```sh
+gcloud compute ssh tfadmin@apigee-cluster-abm-ws0-001 --project=<YOUR_PROJECT> --zone=<YOUR_ZONE>
+```
+
+2. Install the Anthos cluster on the provisioned Compute Engine VM based bare metal infrastructure
+```sh
+sudo ./run_initialization_checks.sh && \
+sudo bmctl create config -c apigee-cluster && \
+sudo cp ~/apigee-cluster.yaml bmctl-workspace/apigee-cluster && \
+sudo bmctl create cluster -c apigee-cluster && \
+```
+
+```
+./install_apigee.sh
+```
+---
+
+Running the commands from the Terraform output starts setting up a new Anthos cluster. This includes checking the initialization state of the nodes, creating the admin and user clusters and also registering the cluster with Google Cloud using [Connect](https://cloud.google.com/anthos/multicluster-management/connect/overview). The whole setup can take up to 30 minutes. You see the following output as the cluster is being created:
+
+> **Note:** The logs for checks on node initialization has been left out. They appear before the following logs from Anthos setup
+
+```sh
```

Comment:
We probably won't need this entire output. I would only the essential parts with ellipses for redundant sections.

```sh
...
Waiting for cluster to become ready OK
Writing kubeconfig file
kubeconfig of created cluster is at bmctl-workspace/apigee-cluster/apigee-cluster-kubeconfig, please run
kubectl --kubeconfig bmctl-workspace/apigee-cluster/apigee-cluster-kubeconfig get nodes
to get cluster node status.
Please restrict access to this file as it contains authentication credentials of your cluster.
Waiting for node pools to become ready OK
Moving admin cluster resources to the created admin cluster
Flushing logs... OK
Deleting bootstrap cluster... OK
...
...
Operation "operations/acat.p2-739559844142-7d29aafc-49d2-47c6-9dd4-4342352c04bd" finished successfully.
Operation "operations/acat.p2-739559844142-79d3808d-6a03-4cbf-ba17-6e8208ecd321" finished successfully.
...
- Processing resources for Istio core.
✔ Istio core installed
- Processing resources for Istiod.
- Processing resources for Istiod. Waiting for Deployment/istio-system/istiod-asm-1106-2
✔ Istiod installed
...
apigeectl_1.7.0-390442d_linux_64/templates/3_apigee-environments.yaml
apigeectl_1.7.0-390442d_linux_64/templates/4_apigee-telemetries.yaml
apigeectl_1.7.0-390442d_linux_64/templates/virtualhosts.yaml
apigeectl_1.7.0-390442d_linux_64/tools/apigee-pull-push.sh
apigeectl_1.7.0-390442d_linux_64/tools/common.sh
apigeectl_1.7.0-390442d_linux_64/tools/create-service-account
apigeectl_1.7.0-390442d_linux_64/tools/dump_kubernetes.sh
apigeectl_1.7.0-390442d_linux_64/apigeectl
/home/tfadmin/apigee_workspace/apigeectl
...
...
Checking Apigee Containers ...
Checking Apigee Containers ...
Checking Apigee Containers ...
Checking Apigee Containers ...
Checking Apigee Containers ...
Apigee is Ready
Anthos on bare metal installation complete!
Run [export KUBECONFIG=/home/tfadmin/bmctl-workspace/apigee-cluster/apigee-cluster-kubeconfig] to set the kubeconfig
Run the [/home/tfadmin/login.sh] script to generate a token that you can use to login to the cluster from the Google Cloud Console
```

Reply:
Fixed

Reply:
Shortened this one.